### DISCUSSION TITLE
The goal of Tracers

### DISCUSSION THREAD

**Discussion by vasigorc** [open-telemetry/opentelemetry-java#3742] [Q&A] [answered]
Hey everyone 👋 
The official Java documentation for OpenTelemetry reads:

> a Tracer must be acquired, which is responsible for creating spans and interacting with the Context. A tracer is acquired by using the OpenTelemetry API specifying the name and version of the library instrumenting the instrumented library or application to be monitored. More information is available in the specification chapter Obtaining a Tracer.

I had hard time understanding [this README section](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#get-a-tracer), specifically whether my application does or doesn't qualify as _instrumented_ or _instrumentation_ library.

It is just an app that has the following dependencies (`sbt` syntax) as copied from [examples/otlp](https://github.com/open-telemetry/opentelemetry-java/blob/main/examples/otlp/build.gradle#L9-L20) project:

```scala
    "io.opentelemetry" % "opentelemetry-api" % otelVersion,
    "io.opentelemetry" % "opentelemetry-extension-annotations" % otelVersion,
    "io.opentelemetry" % "opentelemetry-sdk" % otelVersion,
    "io.opentelemetry" % "opentelemetry-exporter-otlp" % otelVersion,
    "io.opentelemetry" % "opentelemetry-sdk-extension-autoconfigure" % otelInstrumentationVersion,
    "io.opentelemetry" % "opentelemetry-api-metrics" % otelInstrumentationVersion,
    "io.opentelemetry" % "opentelemetry-sdk-metrics" % otelInstrumentationVersion,
    "io.opentelemetry" % "opentelemetry-exporter-otlp-metrics" % otelInstrumentationVersion,
    "io.grpc" % "grpc-netty-shaded" % "1.41.0"
```

I have a service name (set in a  `.properties` file). My application is from high perspective a queue: we listen on different RabbitMQ topics and send each message to a responsible Akka Actor.

I would love to understand where and when should I create a Tracer (or Tracers). Is it once per App? Is it for every high level logical path (business specific action), i.e. per same chain of Spans?  In SdkTraceProvider they seem to be [maintained in memory](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkTracerProvider.java#L36) so I would love to get a better understanding of how/where should I create and manage Traces.

**Comment by jkwatson** [comment, marked as answer]
There is no hard-and-fast rule to these questions. A Tracer is simply a named (and optionally versioned) shim on top of the TracerProvider. The InstrumentationLibraryInfo (name + version + schema) associated with the tracer will be attached to Spans that are generated by the SDK (whether and how that information is exported is up to the individual exporter implementation). If you wish to track the source of your spans at a very granular level, then you can certainly create as many Tracers as you desire, but that is completely up to you.
